modules/Attendance/attendance_take_byCourseClass.php

```diff
@@ -51,20 +54,7 @@
     $gibbonTTDayRowClassID = $_GET['gibbonTTDayRowClassID'] ?? '';
 
     if (empty($gibbonCourseClassID)) {
-        try {
-            $data = array('gibbonPersonID' => $session->get('gibbonPersonID'), 'gibbonSchoolYearID' => $session->get('gibbonSchoolYearID'));
-            $sql = "SELECT gibbonCourseClass.gibbonCourseClassID, gibbonSchoolYear.firstDay, gibbonSchoolYear.lastDay
-                    FROM gibbonCourse
-                    JOIN gibbonSchoolYear ON (gibbonCourse.gibbonSchoolYearID=gibbonSchoolYear.gibbonSchoolYearID)
-                    JOIN gibbonCourseClass ON (gibbonCourseClass.gibbonCourseID=gibbonCourse.gibbonCourseID)
-                    JOIN gibbonCourseClassPerson ON (gibbonCourseClassPerson.gibbonCourseClassID=gibbonCourseClass.gibbonCourseClassID)
-                    WHERE gibbonPersonID=:gibbonPersonID
-                    AND gibbonCourseClass.attendance='Y'
-                    AND gibbonCourse.gibbonSchoolYearID=:gibbonSchoolYearID";
-            $result = $connection2->prepare($sql);
-            $result->execute($data);
-        } catch (PDOException $e) {
-        }
+        $result = $container->get(CourseClassGateway::class)->selectClassByStudent($session->get('gibbonPersonID'), $session->get('gibbonSchoolYearID'));
```

Comment:
This seems to filter only classes with attendance enabled, so I've updated the name. Also, for method order, I tend to go biggest to smallest, to gibbonSchoolYearID is first anytime a method needs it.
```suggestion
        $result = $container->get(CourseClassGateway::class)->selectAttendanceClassesByStudent($session->get('gibbonSchoolYearID'), $session->get('gibbonPersonID'), );
```